openlibrary/templates/lib/nav_head.html

```diff
@@ -98,6 +98,8 @@
           <input type="text" name="q" placeholder="$_('Search')" aria-label="Search" autocomplete="off">
           <input name="mode" type="checkbox" aria-hidden="true" aria-label="Search checkbox" checked="checked" value="" id="ftokenstop" class="hidden instantsearch-mode">
           <input type="submit" value="" class="search-bar-submit" aria-label="Search submit">
+          <div class="vertical-separator"></div>
+          <a href="https://openlibrary.org/barcodescanner?returnTo=/isbn/$$$" id="barcode_scanner_link" class="search-by-barcode-submit" aria-label="$_('Search by barcode')" title="$_('Search by barcode')"></a>
```

Comment:
Oh, also we need to escape the `$` since we want literal `$` :P Replace `$` with `$$`

```suggestion
          <a href="https://openlibrary.org/barcodescanner?returnTo=/isbn/$$$$$$" id="barcode_scanner_link" class="search-by-barcode-submit" aria-label="$_('Search by barcode')" title="$_('Search by barcode')"></a>
```